### PR TITLE
[ADD] webhook: Add standard webhook odoo module

### PR DESCRIPTION
**Pull request by moylop260**

### CODE REVIEW

File: webhook/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+Webhook
+---------------
+
+Module to receive [global webhooks](https://es.wikipedia.org/wiki/Webhook) events.
```
Comment (dreispt) [inline webhook/README.md]: Use link to english Wikipedia?

File: webhook/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+This module invoke methods to process webhook events.
+You will need create a new module to add your logic to process the events with methods called:
+`def run_CONSUMER_EVENT*`
```
Comment (dreispt) [inline webhook/README.md]: Why the star?
Maybe say: "The webhook events will be processed by your custom methods, that should be prefixed with `run_`"

File: webhook/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#
+#    Copyright (c) 2011 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: moylop260@vauxoo.com
+#    planned by: nhomar@vauxoo.com
+#                moylop260@vauxoo.com
+############################################################################
+
+import pprint
```
Comment (dreispt) [inline webhook/controllers/main.py]: remove

File: webhook/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# -*- encoding: utf-8 -*-
+##############################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2011 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
```
Comment (dreispt) [inline webhook/controllers/main.py]: No license info

File: webhook/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+class WebhookController(http.Controller):
+
+    @http.route('/webhook', type='json', auth='none', method=['POST'])
+    def webhook(self, *args, **kwargs):
+        '''
+        Webhook odoo controller to receive json request and send to
+        driver method.
+        You will need create your webhook with http://0.0.0.0:0000/webhook
+        NOTE: Important use --db-filter params in odoo start.
```
Comment (dreispt) [inline webhook/controllers/main.py]: It's important to mention this in the README

File: webhook/__init__.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# -*- encoding: utf-8 -*-
```
Comment (dreispt) [inline webhook/__init__.py]: s/encoding/coding
True for all code files

File: webhook/views/webhook_views.xml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+                </tree>
+            </field>
+        </record>
+        <record model="ir.actions.act_window" id="action_webhook">
+            <field name="name">Webhook</field>
+            <field name="res_model">webhook</field>
+            <field name="view_type">form</field>
+            <field name="view_mode">tree,form</field>
+        </record>
+        <menuitem id="main_webhook_menu" name="Webhook"/>
```
Comment (dreispt) [inline webhook/views/webhook_views.xml]: No parent menu? IMO is should be under Settings / Technical / Automation..

**Comment by dreispt** [comment]
Did some style related comments, but I honestly don't have the technical skill to discuss the semantics of the code.

**Comment by moylop260** [comment]
@dreispt 
Thank for you comment's
I will to work ASAP

**Comment by pedrobaeza** [comment]
This is pending on the creation of the new repo

**Comment by dreispt** [comment]
@pedrobaeza What repo?

**Comment by pedrobaeza** [comment]
webhook repo. It was discussed and it was seen convenient to create it. It's pending from one of our side (board). I will take care of it in the coming days.

**Comment by alan196** [comment]
I just tested this module with odoo 8.0 and 9.0 works fine :+1: 
I made a module with this and it works with no issues.

**Comment by bealdav** [comment]
nice feature, thanks

**Comment by moylop260** [comment]
@bealdav 
You're welcome.

@pedrobaeza @dreispt @gurneyalex 
With the new structure of server-tools: Are there a place for the module of webhooks?

**Comment by pedrobaeza** [comment]
We should move first to create all the repositories, and I think we can include it in any of them for sure.

**Comment by pedrobaeza** [comment]
Should we wait until the code sprint?

**Comment by moylop260** [comment]
Ok, We will discuss on oca sprint
Thanks for reply

**Comment by gurneyalex** [comment]
@moylop260 what is the status of this PR?

**Comment by moylop260** [comment]
This PR add a webhook standard module
In the past, I think that we should be this module with a project separated but I'm considering that is not necessary.
So we could integrate it to server-tools

File: webhook/data/webhook_data.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+        <!--webhook github data-->
+        <record model="webhook" id="webhook_github">
+            <field name="name">github</field>
+            <field name="python_code_get_event">request.httprequest.headers.get('X-Github-Event')</field>
+        </record>
+        <record model="webhook.address" id="webhook_address_github">
+            <field name="name">192.30.252.0/22</field>
+            <field name="webhook_id" ref="webhook_github"/>
+        </record>
```
Comment (sebalix) [inline webhook/data/webhook_data.xml:13]: This GitHub configuration should be part of the demo data, or at least with a ``noupdate="1"``?
We are testing this module, this piece of code is OK excepted the IP address (`192.30.253.0/22` here).

**Review by sebalix — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
@moylop260 are you still interested in this? Should we merge?

**Comment by moylop260** [comment]
Superseded by https://github.com/OCA/server-tools/pull/957
Let's wait a backport work from another contributor...

**Pull request closed without merging**